Tests/TensorFlowTests/LazyTensorTraceTests.swift

```diff
@@ -0,0 +1,151 @@
+import XCTest
+
+@testable import TensorFlow
+import CTensorFlow
+
+final class LazyTensorTraceTests: XCTestCase {
+    override class func setUp() {
+        super.setUp()
+        _RuntimeConfig.useLazyTensor = true
+    }
+
+    func testSingleLiveTensor() {
+        let a = Tensor<Float>(10.0)
+        let b = Tensor<Float>(2.0)
+        let c = Tensor<Float>(3.0)
+        let w = a + b * c
+        XCTAssertEqual(lazyTrace(w)!.description,
+            """
+            lazyTrace_5() -> (Add_4) {
+              Const_0[dtype: float, value: 10.0]():1
+              Const_1[dtype: float, value: 2.0]():1
+              Const_2[dtype: float, value: 3.0]():1
+              Mul_3[T: float](Const_1:0, Const_2:0):1
+              Add_4[T: float](Const_0:0, Mul_3:0):1
+            }
+            """)
+    }
+
+    func testMultipleLiveTensors() {
+        // This test checks that *only* the operations that correspond to `w`,
+        // `y` and `z` are marked as outputs. Specifcally, the intermediate
+        // operations in the trace are not marked as outputs.
+        let a = Tensor<Float>(10.0)
+        let b = Tensor<Float>(2.0)
+        let c = Tensor<Float>(3.0)
+        let w = a + b + c
+        let y = w * c
+        let z = y / (w - c)
+        XCTAssertEqual(lazyTrace(z)!.description,
+            """
+            lazyTrace_8() -> (Add_4,Mul_5,Div_7) {
+              Const_0[dtype: float, value: 10.0]():1
+              Const_1[dtype: float, value: 2.0]():1
+              Add_2[T: float](Const_0:0, Const_1:0):1
+              Const_3[dtype: float, value: 3.0]():1
+              Add_4[T: float](Add_2:0, Const_3:0):1
+              Mul_5[T: float](Add_4:0, Const_3:0):1
+              Sub_6[T: float](Add_4:0, Const_3:0):1
+              Div_7[T: float](Mul_5:0, Sub_6:0):1
+            }
+            """)
+
+        // Note that we only pick operations on which the lazy tensor in
+        // question depends on.
+        XCTAssertEqual(lazyTrace(y)!.description,
+            """
+            lazyTrace_6() -> (Add_4,Mul_5) {
+              Const_0[dtype: float, value: 10.0]():1
```

Comment:
nit: I wish it could be: %0 = Const[...]() Are you the one synthesizing these formats?

Reply:
Good idea. Done.